ginga/rv/plugins/LineProfile.py

```diff
@@ -241,229 +245,373 @@ def redo(self):
+        self.plot.plot(plot_x_axis_data, plot_y_axis_data,
+                       xtitle=self.x_lbl, ytitle=self.y_lbl, marker=marker)
+
+        # Highlight data point from active slice.
+        if i_sel not in (i_x, i_y):
+            i = self.image.revnaxis[i_sel]
+            self.plot.ax.plot(plot_x_axis_data[i], plot_y_axis_data[i],
+                              marker='o', ls='')
+
+        # Top axis to show pixel location across X
+        ax2.cla()
+        xx1, xx2 = self.plot.ax.get_xlim()
+        ax2.set_xlim((xx1 - self._crval) / self._cdelt + self._crpix - 1,
+                     (xx2 - self._crval) / self._cdelt + self._crpix - 1)
+        ax2.set_xlabel('Index')
+        self.plot.draw()
 
     def get_axis(self, i):
         try:
-            self.x_lbl = self.image.get_keyword('CTYPE%d' % i, None)
-            try:
-                kwds = ['CRVAL%d' % i, 'NAXIS%d' % i, 'CDELT%d' % i]
-                crval_i, naxis_i, cdelt_i = self.image.get_keywords_list(*kwds)
+            naxis_s = 'NAXIS{}'.format(i)
+            naxis_i = self.image.get_keyword(naxis_s)
+            self.x_lbl = self.image.get_keyword('CTYPE{}'.format(i), naxis_s)
 
+            try:
+                kwds = ['CRVAL{}'.format(i), 'CDELT{}'.format(i),
+                        'CRPIX{}'.format(i)]
+                crval_i, cdelt_i, crpix_i = self.image.get_keywords_list(*kwds)
             except KeyError as e:
-                raise ValueError("Missing FITS keyword: %s" % str(e))
-
-            axis = crval_i + np.arange(0, naxis_i, 1) * cdelt_i
-
-            if self.x_lbl is not None:
-                units = self.image.get_keyword('CUNIT%d' % i, None)
-                if units is not None:
-                    self.x_lbl += (' (%s)' % str(units))
-            else:
-                self.x_lbl = ''
-            # Assume Y label should always be flux?
-            self.y_lbl = 'Flux'
-            return axis
+                self.logger.error("Missing FITS keyword: {}".format(str(e)))
+                crval_i = 0
+                cdelt_i = 1
+                crpix_i = 1
+
+            n = np.arange(naxis_i) - (crpix_i - 1)
+            axis = crval_i + n * cdelt_i
+            self._crval = crval_i
+            self._cdelt = cdelt_i
+            self._crpix = crpix_i
+
+            units = self.image.get_keyword('CUNIT{}'.format(i), None)
+            if units is not None:
+                self.x_lbl += (' ({})'.format(units))
+
+            # Get pixel value info from header
+            self.y_lbl = self.image.get_keyword('BTYPE', 'Flux')
+            bunit = self.image.get_keyword('BUNIT', None)
+            if bunit is not None:
+                self.y_lbl += (' ({})'.format(bunit))
 
         except Exception as e:
-            errmsg = "Error loading axis %d: %s" % (i, str(e))
+            errmsg = "Error loading axis {}: {}".format(i, str(e))
             self.logger.error(errmsg)
             self.fv.show_error(errmsg)
```

Comment:
Hmm. I thought I removed this but maybe not. Didn't seem to bother me much now anyway.